### ISSUE TITLE
BertTokenizer: Example?

### ISSUE THREAD

**Issue by iLikeToCodeZ** [tensorflow/text#181]
Is there an example on how to use the BertTokenizer?

**Comment by thuang513** [comment]
See example usage in [1] for now. I'm working on a better guide for it and will hopefully have that in soon.

[1] https://github.com/tensorflow/text/blob/master/tensorflow_text/python/ops/bert_tokenizer_test.py

**Comment by hadim** [comment]
I am trying to figure out how to the BERT tokenizer together with BERT modules on TF Hub. Here is what I use right now:

```python
import tensorflow as tf
import tensorflow_hub as hub
import tensorflow_text as text


# Function for preprocessing
# (will probably be part of tensorflow-text soon)
def load_vocab(vocab_file):
    """Loads a vocabulary file into a list."""
    vocab = []
    with tf.io.gfile.GFile(vocab_file, "r") as reader:
        while True:
            token = reader.readline()
            if not token:
                break
            token = token.strip()
            vocab.append(token)
    return vocab


def create_vocab_table(vocab, num_oov=1):
    vocab_values = tf.range(tf.size(vocab, out_type=tf.int64), dtype=tf.int64)
    init = tf.lookup.KeyValueTensorInitializer(keys=vocab, values=vocab_values, key_dtype=tf.string, value_dtype=tf.int64)
    vocab_table = tf.lookup.StaticVocabularyTable(init, num_oov, lookup_key_dtype=tf.string)
    return vocab_table


def merge_dims(rt, axis=0):
    """Collapses the specified axis of a RaggedTensor.

    Suppose we have a RaggedTensor like this:
    [[1, 2, 3],
    [4, 5],
    [6]]

    If we flatten the 0th dimension, it becomes:
    [1, 2, 3, 4, 5, 6]

    Args:
    rt: a RaggedTensor.
    axis: the dimension to flatten.

    Returns:
    A flattened RaggedTensor, which now has one less dimension.
    """
    to_expand = rt.nested_row_lengths()[axis]
    to_elim = rt.nested_row_lengths()[axis + 1]

    bar = tf.RaggedTensor.from_row_lengths(to_elim, row_lengths=to_expand)
    new_row_lengths = tf.reduce_sum(bar, axis=1)
    return tf.RaggedTensor.from_nested_row_lengths(rt.flat_values, rt.nested_row_lengths()[:axis] + (new_row_lengths,))

# Load BERT model from TF Hub
model_url = "https://tfhub.dev/tensorflow/bert_en_cased_L-12_H-768_A-12/1"
bert_layer = hub.KerasLayer(model_url, trainable=False)

# Load vocabulary from BERT TF Hub model
vocab_file = bert_layer.resolved_object.vocab_file.asset_path.numpy()
vocab = load_vocab(vocab_file)

# Check if BERT model is case sensitive
do_lower_case = bert_layer.resolved_object.do_lower_case.numpy()

# Create the vocabulary table
vocab_table = create_vocab_table(vocab, num_oov)

# Define some inputs and tokenize it
text_inputs = ["hello world", 'how are you buddy ?']

tokenizer = text.BertTokenizer(vocab_table, token_out_type=tf.int64, lower_case=do_lower_case)
tokens = tokenizer.tokenize(text_inputs)

# BERT module excepts a 2D tensor (not 3D)
tokens = tokens.to_tensor()[:, :, 0]
tokens = tf.cast(tokens, dtype=tf.int32)

# Set masks and segment ids
input_mask = tf.ones(tokens.shape, dtype=tf.int32)
segment_ids = tf.zeros(tokens.shape, dtype=tf.int32)

# Embed the inputs.
pooled_output, sequence_output = bert_layer([tokens, input_mask, segment_ids])
```

The code runs but I am not sure if the preprocessing I do is enough (I saw some padding with `CLS` and `SEP` in some other issues of this repo). Also, I am not sure I am setting `input_mask` and `segment_ids` correctly for inference.

Hopefully, most of the redundant code here will be integrated into this repo such as `load_vocab()`, `create_vocab_table()` and `merge_dims()`.

**Comment by ejarkm** [comment]
@hadim I recently found that BertTokenizer exists... (After developing the whole thing by myself.

In addition to your code, you could get rid of your load_vocab by simply doing the following:
```python
vocab_file = layer.resolved_object.vocab_file.asset_path # here is the change
tokenizer = text.BertTokenizer(vocab_file)
```
I am aware that you wrote this comment last year but it could help someone else.

**Comment by thuang513** [comment]
As @ejarkm mentioned, you can simplify your code by passing in a path to your vocab file. Also see 
[here](https://blog.tensorflow.org/2020/03/part-1-fast-scalable-and-accurate-nlp-tensorflow-deploying-bert.html) for an example of using BertTokenizer in your TFX pipeline w/ TF.Hub and Transform.
